File: Android/app/src/main/java/com/edu/primary/database/dao/QuestionDao.java
package com.edu.primary.database.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import com.edu.primary.database.entity.QuestionEntity;

import java.util.List;

@Dao
public interface QuestionDao {
    @Insert
    void insertQuestions(List<QuestionEntity> questions);

    @Insert
    long insertQuestion(QuestionEntity question);

    @Query("SELECT * FROM questions WHERE subjectId = :subjectId AND grade = :grade")
    List<QuestionEntity> getQuestionsBySubjectAndGrade(int subjectId, int grade);

    @Query("SELECT * FROM questions WHERE id = :id LIMIT 1")
    QuestionEntity getQuestionById(long id);

    @Query("SELECT COUNT(*) FROM questions WHERE subjectId = :subjectId AND grade = :grade")
    int getQuestionCount(int subjectId, int grade);
}

File: Android-1/app/src/main/java/com/edu/primary/data/local/dao/QuestionDao.kt
package com.edu.primary.data.local.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.edu.primary.data.local.entity.QuestionEntity

@Dao
interface QuestionDao {
    @Insert
    suspend fun insertAll(questions: List<QuestionEntity>)

    @Insert
    suspend fun insert(question: QuestionEntity): Long

    @Query("SELECT * FROM questions WHERE subjectId = :subjectId AND grade = :grade")
    suspend fun getBySubjectAndGrade(subjectId: Int, grade: Int): List<QuestionEntity>

    @Query("SELECT * FROM questions WHERE id IN (:ids)")
    suspend fun getByIds(ids: List<Long>): List<QuestionEntity>

    @Query("SELECT * FROM questions WHERE id = :id LIMIT 1")
    suspend fun getById(id: Long): QuestionEntity?

    @Query("SELECT COUNT(*) FROM questions WHERE subjectId = :subjectId AND grade = :grade")
    suspend fun getCount(subjectId: Int, grade: Int): Int
}
